confgenerator/logging.go

```diff
@@ -70,6 +71,11 @@ func stackdriverOutputComponent(match, userAgent string, storageLimitSize string
 		// Mute these errors until https://github.com/fluent/fluent-bit/issues/4473 is fixed.
 		"net.connect_timeout_log_error": "False",
 	}
+	r := platform.FromContext(ctx).ResourceOverride
+	if r != nil {
```

Comment:
You can construct the `resource_labels` string here from the monitored resources label map in the resource; there's no reason to duplicate that code in each of the resource implementations.

e.g.
```go
if r := platform.FromContext(ctx).ResourceOverride; r != nil {
  mr = r.MonitoredResource()
  var labels []string
  for k, v := range mr.Labels {
    labels = append(labels, fmt.Sprintf("%s=%s", k, v))
  }
  sort.Strings(labels)
  config["resource"] = mr.Type
  config["resource_labels"] = strings.Join(labels, ",")
}
```